### DISCUSSION TITLE
Cannot elevate privileges with polkit

### DISCUSSION THREAD

**Discussion by SionFE** [cockpit-project/cockpit#17648] [Q&A]
Hello. I have Cockpit installed on Gentoo (with systemd). Since this is not officially supported Linux distribution, I don't know whether it is right to ask about the problem in there.
Anyway, the problem is that I cannot elevate privileges with polkit. I get the "spawn failed" error. Configure detects polkit and builds cockpit with support for it. However, it seems that cockpit tries to use sudo that is not installed, because if I remove sudo bridge information from /usr/share/cockpit/shell/manifest.jon I get "No such superuser bridge" error. How to solve this problem without sudo? If I install sudo, cockpit works without issues.

**Comment by jelly** [comment]
Currently Cockpit really only supports sudo in practice.

The next release will include https://github.com/cockpit-project/cockpit/pull/17536 and with that you can specify your own privilege authentication method by overriding the manifest file https://github.com/cockpit-project/cockpit/pull/17536/files#diff-3d973a6f1709fb9f15722e9e3c2f71442c7400f1e2c8524ca6e79125b9b16b6cR280

**Reply by SionFE** [reply]
Updated to 275. What should I change in the manifest file to use polkit for elevating privileges?
Right now I can still use only sudo. Polkit is installed and detected. I also see this message in journal as soon as I log in cockpit:
`Sep 11 02:05:57 KVM polkitd[2470]: Registered Authentication Agent for unix-session:19 (system bus name :1.195 [cockpit-bridge], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8)
Sep 11 02:05:58 KVM dbus-daemon[462]: [system] Activating via systemd: service name='org.freedesktop.hostname1' unit='dbus-org.freedesktop.hostname1.service' requested by ':1.196' (uid=1000 pid=2601 comm="cockpit-bridge" label="kernel")`
